### PR TITLE
Use the real timeToWork values in the top graph on the statistics page

### PR DESCRIPTION
**Pull request by voidus**
This includes a lot of refactoring.
https://github.com/Mayflower/PHProjekt/pull/341 should be merged before this.

### CODE REVIEW

File: phprojekt/htdocs/phpr/Statistics/WorktimeMonthGraph.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        _titleText: function(d) {
+            var date = locale.format(timehelper.dateToJsDate(d.date), {selector: 'date'});
+            if (d.minutesBooked !== 0) {
+                date += ' (' + timehelper.minutesToHMString(d.minutesBooked) + ')';
+            }
+            return date;
+        }
+    });
+
+    var TimeToWorkRenderer = declare(null, {
```
Comment (exi) [inline phprojekt/htdocs/phpr/Statistics/WorktimeMonthGraph.js]: I am a bit confuse as to why this is a declared object although it only has one function that is called like a function and seems to work like a function. does the declaration has any benefit except making the call (new Renderer().render()) more complicated?
I would put all the content into a regular function with a normal function scopy and inline help functions so it is clear how it is used.
Or am i missing something?
Same thing applies to the minutesBookedBlockRenderer

Comment (voidus) [inline phprojekt/htdocs/phpr/Statistics/WorktimeMonthGraph.js]: Fixed

File: phprojekt/htdocs/phpr/Statistics/WorktimeMonthGraph.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                    .attr('y2', horizontalY)
+                    .attr('stroke', '#6aa700');
+        };
+
+        var verticalX1 = function(d, i) {
+            return i * (barPadding + helper.barWidth());
+        };
+
+        var verticalX2 = function(d, i) {
+            return (i) * (barPadding + helper.barWidth());
```
Comment (voidus) [inline phprojekt/htdocs/phpr/Statistics/WorktimeMonthGraph.js]: (i)(i)

File: phprojekt/htdocs/phpr/Statistics/WorktimeMonthGraph.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+                    .append('svg:title')
+                        .text(titleText);
+        };
+    })();
+
+    var renderTimeToWorkLine = (function() {
+        var dayEntries,
+            helper;
+
+        var renderHorizontalLines = function(svgData) {
```
Comment (voidus) [inline phprojekt/htdocs/phpr/Statistics/WorktimeMonthGraph.js]: function renderHorizontalLines(svgData) would be nicer

File: phprojekt/htdocs/phpr/StatisticsView.js [hunk 4]
```diff
@@ -1,140 +1,13 @@
...
     'dijit/_TemplatedMixin',
     'dijit/_WidgetsInTemplateMixin',
-    'phpr/Api',
-    'phpr/Timehelper',
     'dojo/text!phpr/template/statisticsView.html',
-    'd3/d3.v3.js'
-], function(lang, declare, domAttr, locale, all, Widget, Templated, WidgetsInTemplate, api, timehelper,
-            templateString) {
+    'phpr/Statistics/WorktimeMonthGraph',
+], function(declare, Widget, Templated, WidgetsInTemplate, templateString) {
```
Comment (voidus) [inline phprojekt/htdocs/phpr/StatisticsView.js]: one item per line

Comment (voidus) [inline phprojekt/htdocs/phpr/StatisticsView.js]: fixed

File: phprojekt/htdocs/phpr/Statistics/WorktimeMonthGraph.js (hunk 3 above)
Comment (voidus) [inline phprojekt/htdocs/phpr/Statistics/WorktimeMonthGraph.js]: fixed

File: phprojekt/htdocs/phpr/Statistics/WorktimeMonthGraph.js (hunk 2 above)
Comment (voidus) [inline phprojekt/htdocs/phpr/Statistics/WorktimeMonthGraph.js]: fixed

**Pull request merged**